### PR TITLE
Fix VkPhysicalDeviceIDProperties sType validation

### PR DESCRIPTION
**Pull request by krOoze**
`VkPhysicalDeviceIDProperties` is provided from multiple extensions. Add handling of this cornercase into layers. Fix tests that were defective to discover it.

### CODE REVIEW

File: layers/vulkan/generated/stateless_validation_helper.cpp [hunk 1]
```diff
@@ -4184,7 +4185,9 @@ bool StatelessValidation::ValidatePnextPropertyStructContents(const Location& lo
         // Validation code for VkPhysicalDeviceIDProperties structure members
         case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ID_PROPERTIES: {  // Covers VUID-VkPhysicalDeviceIDProperties-sType-sType
 
-            if (!IsExtEnabled(instance_extensions.vk_khr_external_fence_capabilities)) {
+            if (!IsExtEnabled(instance_extensions.vk_khr_external_semaphore_capabilities) &&
```
Comment (spencer-lunarg) [inline layers/vulkan/generated/stateless_validation_helper.cpp:4188]: :100:

**Review by spencer-lunarg — APPROVED** [review]
thanks @krOoze - this is the change I wanted to have, just didn't have time, but this looks awesome and does what it should have done the first time

File: scripts/generators/stateless_validation_helper_generator.py [hunk 2]
```diff
@@ -1001,24 +1003,27 @@ def genStructBody(self, struct: Struct, non_prop_feature: bool):
...
-            else:
-                extension_check = f'if (!IsExtEnabled(instance_extensions.{extension.name.lower()})) {{'
+            extension_conditinals = list()
+            for ext_name in ext_names:
+                extension = self.vk.extensions[ext_name]
+                if extension.device:
+                    extension_conditinals.append( f'((is_physdev_api && !SupportedByPdev(physical_device, vvl::Extension::_{extension.name})) || (!is_physdev_api && !IsExtEnabled(device_extensions.{extension.name.lower()})))' )
+                else:
+                    extension_conditinals.append( f'!IsExtEnabled(instance_extensions.{extension.name.lower()})' )
+            extension_check = f'if ({' && '.join(extension_conditinals)}) {{'
```
Comment (spencer-lunarg) [inline scripts/generators/stateless_validation_helper_generator.py:1026]:
```suggestion
            extension_check = f'if ({" && ".join(extension_conditinals)}) {{'
```

should work... this is a python 3.12 feature I think but we are still at 3.10 python as the minimum requirements

Comment (spencer-lunarg) [reply scripts/generators/stateless_validation_helper_generator.py:1026]: (CI is at 3.10 which is failing on this)

Comment (krOoze) [reply scripts/generators/stateless_validation_helper_generator.py]: I squashed it in

**Pull request merged**